src/index.ts

```diff
@@ -74,6 +84,30 @@ export = class OutputDcatUs11 {
     return searchRequest;
   }
 
+  private extractDatasetSearchRequest(req): IContentSearchRequest {
+    let searchRequest: IContentSearchRequest = null;
+    const id = _.get(req, 'query.id');
+    if (id) {
+      searchRequest = this.getDatasetSearchRequest(id, portalUrl, []);
+    }
+    return searchRequest;
+  }
```

Comment:
I think that passing an ID instead of the request object makes sense here and reduces coupling between this function and the express framework.